drivers/adc/adm1177/adm1177.c

```diff
@@ -0,0 +1,341 @@
+ *
+ * @param cmd - The command byte indicates which operation to be performed.
+ *
+ * @param reg_address - In case of a write extended command byte operation
+ * 			one of the ADM1177_REG_ALERT_EN, ADM1177_REG_ALERT_TH
+ * 			or ADM1177_REG_CONTROL will be used.
+ * 			In case of a write command byte operation reg_address
+ * 			will be set to 0.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_write(struct adm1177_dev *device, uint8_t cmd, uint8_t reg_address)
+{
+	if(reg_address & ADM1177_EXT_CMD_MASK) {
+		uint8_t data[2] = {reg_address, cmd};
+
+		return no_os_i2c_write(device->i2c_desc, data, 2, 1);
+	}
+	if(cmd == ADM1177_CMD_I_CONT || cmd == ADM1177_CMD_I_ONCE) {
+		device->last_command = ADM1177_CURRENT_EN;
+		return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+	}
+	if(cmd == ADM1177_CMD_V_CONT || cmd == ADM1177_CMD_V_ONCE) {
+		device->last_command = ADM1177_VOLTAGE_EN;
+		return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+	}
+	if((cmd == (ADM1177_CMD_I_CONT | ADM1177_CMD_V_CONT)) ||
+	    (cmd == (ADM1177_CMD_I_ONCE | ADM1177_CMD_V_ONCE))) {
+		device->last_command = ADM1177_VOLTAGE_AND_CURRENT_EN;
+		return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+	}
+	return -EINVAL;
+}
+
+/**************************************************************************//**
+ * @brief - Reads the status byte from the ADC if requested and transmits it
+ * 	    through the status_byte pointer.
+ *
+ * @param device - The device structure.
+ *
+ * @param status_byte - The pointer through which the status byte is
+ * 			transmitted.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_read_status(struct adm1177_dev *device, uint8_t *status_byte)
+{
+	int ret;
+	if(device->last_command == ADM1177_STATUS_EN) {
+		if(status_byte == NULL)
+			return -EINVAL;
+
+		ret = no_os_i2c_read(device->i2c_desc, status_byte, 1, 1);
+		if(ret)
+			return ret;
+
+		return 0;
+	}
+	return -EINVAL;
+}
+
+/**************************************************************************//**
+ * @brief - Gets the voltage/current readback from the ADM1177s I2C ADC,
+ *	    if the device is initialized right and transmits the requested
+ *	    values.
+ *
+ * @param device - The device structure.
+ *
+ * @param conv_voltage - The variable through which the voltage read from the
+ * 			 ADC is received if requested.
+ * 			 If it is 0, it means it was not requested.
+ *
+ * @param conv_current - The variable through which the current read from the
+ * 			 ADC is received if requested.
+ * 			 If it is 0, it means it was not requested.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_read(struct adm1177_dev *device,
```

Comment:
`adm1177_read` means that it is a generic read function usable for any kind of register reads. which is not the case here. a more specific header should be used (same as we have for `adm1177_read_status`).

Reply:
other than readback, which was the previous name for this read function doesn't come to mind after reading the data sheet.